wezterm-input-types/src/lib.rs

```diff
@@ -1808,31 +1820,32 @@ impl KeyEvent {
                     *shifted_key
                 };
 
-                if !flags.contains(KittyKeyboardFlags::REPORT_ALTERNATE_KEYS)
+                let use_legacy = is_legacy_key
                     && event_type.is_empty()
-                    && is_legacy_key
-                    && !(flags.contains(KittyKeyboardFlags::DISAMBIGUATE_ESCAPE_CODES)
-                        && (self.modifiers.contains(Modifiers::CTRL)
-                            || self.modifiers.contains(Modifiers::ALT)))
-                {
-                    use_legacy = true;
-                }
+                    && !flags.contains(KittyKeyboardFlags::REPORT_ALTERNATE_KEYS)
+                    && !flags.contains(KittyKeyboardFlags::DISAMBIGUATE_ESCAPE_CODES);
 
                 if use_legacy {
                     // Legacy text key
+                    // https://sw.kovidgoyal.net/kitty/keyboard-protocol/#legacy-text-keys
+
                     let mut output = String::new();
                     if self.modifiers.contains(Modifiers::ALT) {
                         output.push('\x1b');
                     }
+
                     if self.modifiers.contains(Modifiers::CTRL) {
                         csi_u_encode(
                             &mut output,
-                            shifted_key.to_ascii_uppercase(),
+                            shifted_key.to_ascii_uppercase(), // Is this to uppercase intended?
```

Comment:
@wez : Is this intended to turn it into uppercase here? Or is this a typo/mistake where one wanted to tackle point 3 in the protocol : https://sw.kovidgoyal.net/kitty/keyboard-protocol/#legacy-text-keys